packages/lib/src/project/AppProject.ts

```diff
@@ -63,18 +65,53 @@ class BaseAppProject extends BasePackageProject {
     version = semanticVersionToString(version);
 
     try {
-      app = appAddress
-        ? await App.fetch(appAddress, txParams)
-        : await App.deploy(txParams);
-      if (packageAddress) thepackage = Package.fetch(packageAddress, txParams);
-      else if (await app.hasPackage(name, version))
+      if (appAddress) {
+        app = await App.fetch(appAddress, txParams);
+      } else {
+        Loggy.add(
+          `${fileName}#fetchOrDeploy`,
+          `publish-project`,
+          'Preparing everything to publish the project! Deploying new App contract',
+        );
+        app = await App.deploy(txParams);
+      }
+
+      if (packageAddress) {
+        thepackage = Package.fetch(packageAddress, txParams);
+      } else if (await app.hasPackage(name, version)) {
         thepackage = (await app.getPackage(name)).package;
-      else thepackage = await Package.deploy(txParams);
-      directory = (await thepackage.hasVersion(version))
-        ? await thepackage.getDirectory(version)
-        : await thepackage.newVersion(version);
+      } else {
+        Loggy.addOrUpdate(
+          `${fileName}#fetchOrDeploy`,
+          `publish-project`,
+          'Deploying new Package contract',
+        );
+        thepackage = await Package.deploy(txParams);
+        Loggy.update(`publish-project`, {
+          text: 'Deploying new Package contract',
```

Comment:
Isn't this updating the spinner with the same text as before?

Reply:
Good catch!